### PR TITLE
New recipes: equality, ordering, numbers

### PR DESCRIPTION
**Pull request by grossbart**
I'm still trying to find the right tone, content, and structure, so these might be adjusted in the future. However, they've shown me some things that would need to be covered in a tutorial, which I might start writing next.

Because these are "recipes", I tried to keep the explanations to a minimum: the code should be copy-pasteable and not more at this point. Explanations will follow in different sections of the docs.

Some examples are already well documented in, say, the `Option` documentation page, but for people not knowing where to look those would be unfindable. This is why recipes provide an alternative entry point. Let's see where this goes …

### CODE REVIEW

File: docs/recipes/equality.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+setoidUserId.equals({ userId: 1, name: 'Giulio' }, { userId: 1, name: 'Giulio Canti' }) // true
+setoidUserId.equals({ userId: 1, name: 'Giulio' }, { userId: 2, name: 'Giulio' }) // false
+```
+
+## More Setoid instances
+
+Many data types provide `Setoid` instances. Here's [Option](../modules/Option.ts):
+
+```ts
+import { getSetoid, none, some } from '../lib/Option'
```
Comment (gcanti) [inline docs/recipes/equality.md]: typo: `../lib` -> `fp-ts/lib`

File: docs/recipes/equality.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import { setoidNumber } from '../lib/Setoid'
```
Comment (gcanti) [inline docs/recipes/equality.md]: idem

File: docs/recipes/equality.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+O.equals(some(3), some(3)) // true
+O.equals(none, some(4)) // false
+O.equals(none, none) // true
+```
+
+It works similarly for [Either](../modules/Either.ts) and other types where it is possible to determine equality:
+
+```ts
+import { getSetoid, left, right } from '../lib/Either'
```
Comment (gcanti) [inline docs/recipes/equality.md]: idem

File: docs/recipes/equality.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import { setoidNumber, setoidString } from '../lib/Setoid'
```
Comment (gcanti) [inline docs/recipes/equality.md]: idem

File: docs/recipes/ordering.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+const O = getOrd(ordNumber)
+O.compare(none, none) // 0
+O.compare(none, some(1)) // -1
+O.compare(some(1), none) // 1
+O.compare(some(1), some(2)) // -1
+O.compare(some(1), some(1)) // 0
+```
+
+It works similarly for [Tuple](../modules/Tuple.ts)`s and other types where it is possible to determine order:
```
Comment (gcanti) [inline docs/recipes/ordering.md]: It's me or the diff display is weird here, maybe that additional "`"?

File: docs/recipes/equality.md (hunk 1 above)
Comment (grossbart) [reply docs/recipes/equality.md]: Thanks, last minute copy-pasting …

**Pull request merged**